src/poetry/console/commands/show.py

```diff
@@ -130,6 +130,10 @@ def handle(self) -> Optional[int]:
         if self.option("default"):
             only_groups.append("default")
 
+        if not self.poetry.locker.is_locked():
+            self.line_error("<comment>The lock file does not exist. Locking.</comment>")
+            self.call("lock")
```

Comment:
IMO `poetry show` shouldn't have the side effect of locking. A message about the missing lock file and maybe an instruction on what to do now should be sufficient.

Reply:
Yea I went back and forth on that as well, I'm fine either way 👍